FastBank.Services/BankAccountService/BankAccountService.cs

```diff
@@ -105,9 +110,113 @@ public void WithdrawAmount(Customer customer, BankAccount customerBankAccount)
+                {
+                    Console.WriteLine("Plese input correct ammount to transfer (press any key to continue...)");
+                    var keyIsEnter = Console.ReadKey();
+                    new MenuService().MoveToPreviousLine(keyIsEnter, 3);
+                }
+            }
+            while (amountToTransfer <= 0);
+
+            if (amountToTransfer > 0)
+            {
+                bool hasEnoughFunds = (customerBankAccount.Amount - amountToTransfer) < 0;
+                if (hasEnoughFunds)
+                {
+                    Console.WriteLine("You do not have enough funds to transfer (press any key to continue...)");
+                    Console.ReadKey();
+                }
+                else
+                {
+                    var friendAccount = bankAccountRepository.GetBankAccountByCustomer(new Customer(friend));
+                    if (friendAccount == null)
+                    {
+                        Console.WriteLine("You friend has not bank account. Press any key to continue...");
+                        Console.ReadKey();
+                    }
+                    else
+                    {
+                        Console.WriteLine($"Please confirm with Y transfer of {amountToTransfer} to {friend.Name} or press any other key to cancel...");
+                        var confirmKey = Console.ReadKey();
+                        if (confirmKey.KeyChar == 'Y')
+                        {
+                            friendAccount.DepositAmount(amountToTransfer);
+                            Update(friendAccount);
+                            customerBankAccount.WithdrawAmount(amountToTransfer);
+                            Update(customerBankAccount);
+                        }
+                    }
+                }
+            }
+        }
+
+        public void TransferMoneyToFriendMenu(Customer customer, BankAccount customerBankAccount)
         {
-            throw new NotImplementedException();
+            Console.Clear();
+            _menuService.Logo();
+
+            Dictionary<int, User> friends = new Dictionary<int, User>();
+            var friendIndex = 0;
+            var friendsList = _userService.GetUserFriends(customer);
+            if (friendsList != null)
+            {
+
+                Console.WriteLine($"\nYou have {friendsList.Count} friend{(friendsList.Count > 1 ? 's' : string.Empty)}:\n");
+                foreach (var friend in friendsList)
+                {
+                    friends.Add(++friendIndex, friend);
+                    Console.WriteLine($"{friendIndex}. Name: {friend.Name}; email: {friend.Email}");
+                }
+            }
+
+            var menuOptions = $"\nPlease choose your action: " +
+                             $"\n1: Add friend. 2. Remove friend. 3. Transfer money to friend  0: for exit";
+            int action = _menuService.CommandRead(new Regex("^[0123]{1}$"), menuOptions);
```

Comment:
_menuService.CommandRead(new Regex("^[0123]{1}$"), menuOptions); 
This code has been repeated a lot. Please create a method which will have regex in it defined and only you will need to pass menuOptions as parameter

Reply:
DRY principle

Reply:
Fix by this way CommandRead(int countOfOptions, string menuOptions)